src/api/getAuthRequest.js

```diff
@@ -2,15 +2,32 @@ import config from '../config'
 
 import { getToken } from './getToken'
 
-export const getAuthRequest = async path => {
+export const getAuthRequest = async (path, type) => {
   const { access_token } = await getToken()
   const res = await fetch(`${config.apiUrl}/api/${path}`, {
+    method: 'GET',
     headers: {
       'Content-Type': 'application/json',
       Authorization: `Bearer ${access_token}`,
     },
   })
 
-  const json = await res.json()
-  return json
+  const products = await res.json()
+
+  switch (type) {
+    case 'products':
```

Comment:
this is not right place where you want to transform received data. If you need to make some custom transformations, make it after data are fetched and before setting new state of component with that data. This "transformer" can be standalone function because of reusability reasons (you might want to use it anywhere else) and also because of code readability (not to have componentDidMount lifecycle very long)

Why this is not good solution? When there'll be more requests (and also transformation requirements for that requests), this file will grow and can become unreadable

Reply:
I thought that you will not agree with this, but actually i dont know how to handle this different way. 
Okay, fixed!